datajs/src/file-base.js

```diff
@@ -140,13 +142,35 @@ export class FileInterface extends File {
     return this.descriptor.name
   }
 
-  async hash() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('md5').update(text).digest('hex')
-  }
-
-  async hashSha256() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('sha256').update(text).digest('hex')
+  async generateHash(hashType, cbProgress) {
+    return new Promise((resolve, reject) => {
+      let newHash =  hashType === "md5" ? crypto.createHash('md5') : crypto.createHash('sha256');
+      readChunked(this.descriptor, (chunk, offs, total) => {
+        newHash.update(chunk);
+        if (cbProgress) {
+          cbProgress(offs / total);
+        }
+      }, err => {
+        if (err) {
+          reject(err);
+        } else {
+          // TODO: Handle errors
+          let hashHex = newHash.digest('hex')
+          resolve(hashHex);
+        }
+      });
+    });
+  }
+
+  // call content function and generate MD5 hash from this content
+  // tested with big file 1.9Gb
+  async hash(cbProgress) {
```

Comment:
Please add JS Doc here and below function as well expalining also what is `cbProgress`

Reply:
Done